plugins/orchestrator-backend/src/service/router.ts

```diff
@@ -494,115 +494,120 @@ function setupInternalRoutes(
-        );
-        res
-          .status(500)
-          .send(`Couldn't fetch workflow definition ${workflowId}`);
-        return;
-      }
-
-      const serviceUrl = workflowDefinition.serviceUrl;
-      if (!serviceUrl) {
-        auditLogRequestError(
-          new Error(`Service URL is not defined for workflow ${workflowId}`),
-          endpointName,
-          endpoint,
-          req,
-        );
-        res
-          .status(500)
-          .send(`Service URL is not defined for workflow ${workflowId}`);
-        return;
-      }
-
-      const definition =
-        await services.orchestratorService.fetchWorkflowDefinition({
-          definitionId: workflowId,
-          cacheHandler: 'throw',
-        });
-
-      if (!definition) {
-        auditLogRequestError(
-          new Error(
-            `Couldn't fetch workflow definition of workflow source ${workflowId}`,
-          ),
-          endpointName,
-          endpoint,
-          req,
-        );
-        res
-          .status(500)
-          .send(
-            `Couldn't fetch workflow definition of workflow source ${workflowId}`,
+      try {
+        const workflowDefinition =
+          await services.orchestratorService.fetchWorkflowInfo({
+            definitionId: workflowId,
+            cacheHandler: 'throw',
+          });
+
+        if (!workflowDefinition) {
+          auditLogRequestError(
+            new Error(`Couldn't fetch workflow definition ${workflowId}`),
+            endpointName,
+            endpoint,
+            req,
           );
-        return;
-      }
-
-      if (!definition.dataInputSchema) {
-        res.status(200).json({});
-        return;
-      }
+          res
+            .status(500)
+            .send(`Couldn't fetch workflow definition ${workflowId}`);
+          return;
+        }
+
+        const serviceUrl = workflowDefinition.serviceUrl;
+        if (!serviceUrl) {
+          auditLogRequestError(
+            new Error(`Service URL is not defined for workflow ${workflowId}`),
+            endpointName,
+            endpoint,
+            req,
+          );
+          res
+            .status(500)
+            .send(`Service URL is not defined for workflow ${workflowId}`);
+          return;
+        }
 
-      const instanceVariables = instanceId
-        ? await services.orchestratorService.fetchInstanceVariables({
-            instanceId,
+        const definition =
+          await services.orchestratorService.fetchWorkflowDefinition({
+            definitionId: workflowId,
             cacheHandler: 'throw',
-          })
-        : undefined;
-
-      const workflowData = instanceVariables
-        ? services.dataInputSchemaService.extractWorkflowData(instanceVariables)
-        : undefined;
-
-      const workflowInfo = await routerApi.v2
-        .getWorkflowInputSchemaById(workflowId, serviceUrl)
-        .catch((error: { message: string }) => {
-          auditLogRequestError(error, endpointName, endpoint, req);
```

Comment:
cayou explain how this change is related to error handling? I'm getting lost

Reply:
all V2 functions catch the errors, call audit and return result. only this API did not do it. some of the code is outside the call to the V2 object so we had to wrap it all with try-catch.

Reply:
we can't remove try catch from sonataflow service unless all the calls to it in the router catch the errors. this was not the case with /inputSchema API so Yaron fixed it